### DISCUSSION TITLE
path method for ui.file

### DISCUSSION THREAD

**Discussion by Buzzardarg** [marimo-team/marimo#2083] [Ideas]
Hello, I was wondering if there is any way to get the path for files selected using ui.file and in case that it's not possible, can a path method be added for this purpose?

thanks!

**Comment by mscolnick** [comment]
Hi @Buzzardarg, it's not possible since `mo.ui.file` uses the browser file upload API, which does not give you paths (likely for security reasons). 

You can you `mo.ui.file_browser` which uses the backend as a file-browser, and does give you access to the file paths. 

If you want something that is a blend between them, we would welcome a contribution for adding some support
